nixos/doc/manual/release-notes/rl-1903.xml

```diff
@@ -211,15 +211,76 @@
      and options <literal>hardware.ckb.*</literal> are renamed to
      <literal>hardware.ckb-next.*</literal>.
     </para>
-   </listitem>
+  </listitem>
    <listitem>
     <para>
      The option <literal>services.xserver.displayManager.job.logToFile</literal> which was
      previously set to <literal>true</literal> when using the display managers
      <literal>lightdm</literal>, <literal>sddm</literal> or <literal>xpra</literal> has been
      reset to the default value (<literal>false</literal>).
     </para>
+  </listitem>
+  <listitem>
+      The <literal>nscd</literal> now disables all caching of
+      <literal>passwd</literal> and <literal>group</literal> databases by
+      default. This was interferring with the correct functioning of the
+      <literal>libnss_systemd.so</literal> module which is used by
+      <literal>systemd</literal> to manage uids and usernames in the presence
+      of <literal>DynamicUser=</literal> in systemd services.
+      The was already the default behaviour in presence of
+      <literal>services.sssd.enable = true</literal> because nscd caching
+      would interfere sssd in unpredictable ways as well.Because we're using nscd
+      not for caching, but for convincing glibc to find NSS modules in the
+      nix store instead of an absolute path, we have decided to disable
+      caching globally now, as it's usually not the behaviour the user wants
+      and can lead to surprising behaviour.
+      Furthermore, negative caching of host lookups is also disabled now by
+      default. This should fix the issue of dns lookups failing in the
+      presence of an unreliable network.
```

Comment:
It occurs to me that this could be fixed by calling `nscd --invalidate` (or `systemctl reload nscd`) when the network configuration changes. I seem to remember that we used to do that, but I'm not sure.

Reply:
This would be require to be implement in dhcpcd, network-manager and networkd.
This seems too complex to me for backporting and would need proper testing.

Reply:
Either way this shouldn't be backported because it could have significant performance implications.

Reply:
It should be still disabled if resolved is enabled, because that one is not affected by `resolfconf` and caches hosts itself.